### PR TITLE
Implement Text onLongPress

### LINKED ISSUE #5855: Missing `onLongPress` for `<Text/>`
For [Touchable*](http://facebook.github.io/react-native/docs/touchablewithoutfeedback.html#props) components, there are `onPress` and `onLongPress` props, but for [Text](http://facebook.github.io/react-native/docs/text.html), there's only one `onPress` prop :cry: 

For my use case, I'm using `<Text/>` that are styled like hyperlinks.

### PR DESCRIPTION
**Pull request by grabbou**
Fixes #5855

Tested with `UIExplorer`, first <Text> is the `onPress`, second is the `onLongPress`.

### CODE REVIEW

File: Libraries/Text/Text.js [hunk 1]
```diff
@@ -154,7 +158,7 @@ const Text = React.createClass({
                 }
               }
               this.touchableHandleActivePressIn = () => {
-                if (this.props.suppressHighlighting || !this.props.onPress) {
+                if (this.props.suppressHighlighting || !this.props.onPress && !this.props.onLongPress) {
```
Comment (mkonicek) [inline Libraries/Text/Text.js]: I find this condition a bit hard to read. Can you add brackets to show how this logically belongs together or maybe split it into two return statements?

File: Libraries/Text/Text.js [hunk 2]
```diff
@@ -163,7 +167,7 @@ const Text = React.createClass({
               };
 
               this.touchableHandleActivePressOut = () => {
-                if (this.props.suppressHighlighting || !this.props.onPress) {
+                if (this.props.suppressHighlighting || !this.props.onPress && !this.props.onLongPress) {
```
Comment (mkonicek) [inline Libraries/Text/Text.js]: Same here.

**Comment by mkonicek** [comment]
Good idea!

File: Libraries/Text/Text.js (hunk 1 above)
Comment (grabbou) [inline Libraries/Text/Text.js]: Sounds reasonable to me. I am thinking it could just be `const hasPressHandlers` instead. What you think?

**Comment by grabbou** [comment]
@mkonicek I think we are good here, extracted all these things into a single function to make it more readable.

File: Libraries/Text/Text.js [hunk 3]
```diff
@@ -124,6 +128,9 @@ const Text = React.createClass({
   contextTypes: {
     isInAParentText: React.PropTypes.bool
   },
+  hasPressHandler(): boolean {
```
Comment (janicduplessis) [inline Libraries/Text/Text.js]: Can you prefix the method with an `_` otherwise it will show up in the docs on the website.

Comment (grabbou) [inline Libraries/Text/Text.js]: Done, thanks!

**Comment by mkonicek** [comment]
Find it much more readable now, thanks @grabbou!

@facebook-github-bot shipit

**Comment by grabbou** [comment]
Flow errors fixed, re-trying.

**Comment by grabbou** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**